profile.d/00-pengwin.sh

```diff
@@ -1,5 +1,22 @@
-# enable external x display
-export DISPLAY=:0
+# check whether it is WSL1 for WSL2 by using gcc version kernel built with
+# check whether we have wsl.exe in path
+if ( which wsl.exe >/dev/null ) && ( $(which wsl.exe) -l -v >/dev/null ) && [ $(wsl.exe -l -v 2>&1 |sed -e "s|\r||g" -e "s|\x00||g" | grep WLinux | awk '{print $3}') -eq 2]; then
+  # enable external x display for WSL 2
+
+  ipconfig_exec=$(wslpath "C:\\Windows\\System32\\ipconfig.exe")
+  if ( which ipconfig.exe >/dev/null ); then
+    ipconfig_exec=$(which ipconfig.exe)
+  fi
+
+  wsl2_d_tmp="eval "$ipconfig_exec" | grep -n WSL | cut -d : -f 1)"
```

Comment:
In the grep it should look for  the IP which has the Default Gateway not the WSL one because it doesn't work in every case

Reply:
@crramirez I am a bit confused here. What do you mean here?

Reply:
This is why: `export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0` doesn't work in every case because it give the ip of the WSL VM and XServer not always bind to this address. Hopefully, in my configuration it happens so I can replicate. 

So, instead of using this ip, it is better to use the one which has the default gateway, it means the one which has the internet connection.

Check the code I put in the comment. It has the change. And test it in your configuration

Reply:
Okay, I get it now. It is indeed better after testing.